packages/api/src/graphql/article/article.queries.ts

```diff
@@ -58,6 +58,84 @@ const createTitleFilter = (filter: Partial<ArticleFilter>): Prisma.ArticleWhereI
   return {}
 }
 
+const createPreTitleFilter = (filter: Partial<ArticleFilter>): Prisma.ArticleWhereInput => {
+  if (filter?.preTitle) {
+    const containsPreTitle: Prisma.ArticleRevisionWhereInput = {
+      preTitle: {
+        contains: filter.preTitle,
+        mode: 'insensitive'
+      }
+    }
+
+    return {
+      OR: [{draft: containsPreTitle}, {pending: containsPreTitle}, {published: containsPreTitle}]
+    }
+  }
+
+  return {}
+}
+
+const createLeadFilter = (filter: Partial<ArticleFilter>): Prisma.ArticleWhereInput => {
```

Comment:
I would consider writing a generic filter function at least for string filters. Otherwise this is just unnecessary code duplication.